### PR TITLE
Several "remote branch" should be "remote-tracking branch".

### PR DESCRIPTION
**Pull request by danny0838**
Also added some text to introduce remote references. Ref. #301.

### CODE REVIEW

File: book/03-git-branching/sections/remote-branches.asc [hunk 1]
```diff
@@ -2,9 +2,13 @@
...
-Remote branches are references (pointers) to the state of branches in your remote repositories.
-They're local branches that you can't move; they're moved automatically for you whenever you do any network communication.
-Remote branches act as bookmarks to remind you where the branches on your remote repositories were the last time you connected to them.
+Remote references are references (pointers) in your remote repositories, including branches, tags, and so on.
+You can get a full list of remote references explicitly with `git ls-remote (remote)`, or `git remote show (remote)` for remote branches as well as more information.
+Nevertheless, a more common way is to take the advantage of remote-tracking branches.
+
+Remote-tracking branches are references to the state of remote branches.
+They're local references that you can't move; they're moved automatically for you when you interact with the corresponding remote repository.
+Remote-tracking act as bookmarks to remind you where the branches in your remote repositories were the last time you connected to them.
```
Comment (ben) [inline book/03-git-branching/sections/remote-branches.asc]: This should be "Remote-tracking branches".

File: book/03-git-branching/sections/remote-branches.asc [hunk 2]
```diff
@@ -2,9 +2,13 @@
 === Remote Branches
 
 (((branches, remote)))(((references, remote)))
-Remote branches are references (pointers) to the state of branches in your remote repositories.
-They're local branches that you can't move; they're moved automatically for you whenever you do any network communication.
-Remote branches act as bookmarks to remind you where the branches on your remote repositories were the last time you connected to them.
+Remote references are references (pointers) in your remote repositories, including branches, tags, and so on.
+You can get a full list of remote references explicitly with `git ls-remote (remote)`, or `git remote show (remote)` for remote branches as well as more information.
+Nevertheless, a more common way is to take the advantage of remote-tracking branches.
```
Comment (ben) [inline book/03-git-branching/sections/remote-branches.asc]: "take advantage"

**Comment by ben** [comment]
Just a couple of small grammar fixes. Thanks for tracking these down!

**Comment by YueLinHo** [comment]
:+1:

**Comment by ben** [comment]
Brilliant. Thanks!

**Pull request merged**